### ISSUE TITLE
NPP v8.2.2 crash

### ISSUE THREAD

**Issue by pryrt** [pryrt/Win32-Mechanize-NotepadPlusPlus#84]
Per @vinsworldcom, https://community.notepad-plus-plus.org/post/73556, 

    editor->getTextRange(MIN, MAX)

crashes Notepad++ 8.2.2

**Comment by vinsworldcom** [comment]
@pryrt 

quick test on my installed 0.08 version and no crash!  All did was change `ll` to `qq` in 'Editor.pm':

```
324c324
<     my $packed_struct = pack "ll $pk", $start, $end, $text_buf->{ptr};
---
>     my $packed_struct = pack "qq $pk", $start, $end, $text_buf->{ptr};
327c327
<         my ($smin,$smax,$ptr) = unpack "ll $pk", $packed_struct;
---
>         my ($smin,$smax,$ptr) = unpack "qq $pk", $packed_struct;
```

mileage may vary if you need to support 32-bit as well since PerlDoc shows:

```
> perldoc -f pack
[...[

                l  A signed long (32-bit) value.
                L  An unsigned long value.

                q  A signed quad (64-bit) value.
                Q  An unsigned quad value.
                     (Quads are available only if your system supports 64-bit
                      integer values _and_ if Perl has been compiled to support
                      those.  Raises an exception otherwise.)
[...]
```

Cheers.

**Comment by pryrt** [comment]
@vinsworldcom , thanks for that test.

I think my code will have to have logic checking for 64-bit _and_ NPP>=v8.2.2, because if I understand it correctly, if I use an older version of Notepad++, it will still require the `ll` instead of `qq`.

Also: `getStyledText`, and possibly `findText`, have similar structures and probably need the same treatment.

**Comment by vinsworldcom** [comment]
Unfortunately, you're probably correct on all counts!

Good news is as with python script it only crashes N++ if you actually call one of those two methods. I don't use them in any of my scripts.

Cheers.

**Comment by pryrt** [comment]
closed by [b957dc7](https://github.com/pryrt/Win32-Mechanize-NotepadPlusPlus/commit/b957dc7b00595e3e6f6c28681328058205d99641) and released in https://github.com/pryrt/Win32-Mechanize-NotepadPlusPlus/commit/67a333a530cd20610aa79d2cb14078e37168ca34
